collector/src/descriptor/kind/bridge_network_status.rs

```diff
@@ -0,0 +1,199 @@
+            },
+           uniq("flag-thresholds") [] => {
+                // TODO should reject when split_once fail
+                flags: rest.iter()
+                            .filter_map(|v| v.split_once('='))
+                            .map(|(k,v)| (k.to_owned(), v.to_owned()))
+                            .collect(),
+            },
+            uniq("fingerprint") [] => {
+                fingerprint: rest.join(" "),
+            },
+
+        }})
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Address {
+    Ipv4(SocketAddrV4),
+    Ipv6(SocketAddrV6),
+}
+
+impl Address {
+    fn parse(input: &str) -> Result<Self, Error> {
+        if let Ok(addr) = input.parse::<SocketAddrV4>() {
+            Ok(Self::Ipv4(addr))
+        } else {
+            Ok(Self::Ipv6(input.parse::<SocketAddrV6>()?))
+        }
+    }
+}
+
+#[derive(Debug, Builder, Clone)]
+pub struct NetworkStatus {
+    pub nickname: String,
+    pub identity: String,
+    pub digest: String,
+    pub publication: DateTime<Utc>,
+    pub ipv4: Ipv4Addr,
+    pub or_port: u16,
+    pub dir_port: u16,
+    #[builder(setter(custom))]
+    pub addresses: Vec<Address>,
+    pub flags: Vec<String>,
+    pub bandwidth: u64,
+    #[builder(setter(custom))]
+    pub policies: Vec<Policy>,
+}
+
+impl NetworkStatusBuilder {
+    fn addresses(mut self, value: Address) -> Self {
+        match self.addresses {
+            Some(ref mut addr) => addr.push(value),
+            None => self.addresses = Some(vec![value]),
+        }
+        self
+    }
+
+    fn policies(mut self, value: Policy) -> Self {
+        match self.policies {
+            Some(ref mut pol) => pol.push(value),
+            None => self.policies = Some(vec![value]),
+        }
+        self
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Policy {
+    Accept(String),
+    Reject(String),
+}
+
+#[derive(Debug)]
+pub struct BridgeNetworkStatus {
+    pub header: Header,
+    pub network_status: Vec<NetworkStatus>,
+}
+
+impl BridgeNetworkStatus {
+    pub fn parse(input: &str, version: (u32, u32)) -> Result<Self, Error> {
+        use crate::descriptor::nom_combinators::*;
+
+        if version.0 != 1 || version.1 > 2 {
+            return Err(ErrorKind::UnsupportedDesc(format!(
+                "bridge-network-status v{}.{} is not supported",
+                version.0, version.1
+            ))
+            .into());
+        }
+
+        let header = Header::parse(&format!("{}\n", input.lines().take(3).collect::<Vec<_>>().join("\n")))?;
+
+        let mut network_status = Vec::new();
+
+        input.lines().skip(3).fold(
+            Ok(NetworkStatusBuilder::default()),
+            |acc, line| -> Result<NetworkStatusBuilder, Error> {
+                match parse_line(line) {
+                    ("r", params) => {
+                        let mut builder = acc?;
+
+                        if let Ok(net) = builder.build() {
```

Comment:
can't that silence errors on some invalid status, Like one missing a bandwidth line?